tests/data/test1082

```diff
@@ -32,7 +32,7 @@ http
 HTTP GET with localhost --interface
  </name>
  <command>
-http://%HOSTIP:%HTTPPORT/%TESTNUMBER -4 --interface localhost
+http://%HOSTIP:%HTTPPORT/%TESTNUMBER -4 --interface 127.0.0.1
```

Comment:
I could not reproduce the test error locally. I assume because the interface whose IPv6 address is [::1] on the test machines is different from interface of 127.0.0.1, and since localhost resolved to [::1], the socket was bound to the wrong interface.

Maybe this is a good reason to keep `CURLOPT_IPRESOLVE` separated from the new option `CURLOPT_IP_PROTO`?